lib/plugins/aws/package/compile/events/s3/index.js

```diff
@@ -10,9 +10,32 @@ class AwsCompileS3Events {
     this.options = options;
     this.provider = this.serverless.getProvider('aws');
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8031
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 's3', {
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      anyOf: [
+        { $ref: '#/definitions/awsS3BucketName' },
+        {
+          type: 'object',
+          properties: {
+            bucket: { $ref: '#/definitions/awsS3BucketName' },
```

Comment:
We should support here any string, as it could be reference to bucket data in `provider.s3` group (and over there might be an actual _name_ for bucket configured)